src/lib.rs

```diff
@@ -593,49 +592,28 @@ impl Iterator for Masks {
 ///
 /// This struct is created by the [`FixedBitSet::ones`] method.
 pub struct Ones<'a> {
-    current_bit_idx: usize,
-    current_block_idx: usize,
+    bitset: Block,
+    block_idx: usize,
     remaining_blocks: &'a [Block],
-    current_block: Block
 }
 
 impl<'a> Iterator for Ones<'a> {
     type Item = usize; // the bit position of the '1'
 
     #[inline]
     fn next(&mut self) -> Option<Self::Item> {
-        let mut block = self.current_block;
-        let mut idx = self.current_bit_idx;
-
-        loop {
-            loop {
-                if (block & 1) == 1 {
-                    self.current_block = block >> 1;
-                    self.current_bit_idx = idx + 1;
-                    return Some(idx);
-                }
-                // reordering the two lines below makes a huge (2x) difference in performance!
-                block = block >> 1;
-                idx += 1;
-                if block == 0 {
-                    break;
-                }
-            }
-
-            // go to next block
-            match self.remaining_blocks.split_first() {
-                Some((&next_block, rest)) => {
-                    self.remaining_blocks = rest;
-                    self.current_block_idx += 1;
-                    idx = self.current_block_idx * BITS;
-                    block = next_block;
-                }
-                None => {
-                    // last block => done
-                    return None;
-                }
+        while self.bitset == 0 {
+            if self.remaining_blocks.is_empty() {
```

Comment:
We can accomplish all of `is_empty, [0] and [1..]` in one method call - `self.remaining_blocks.split_first()` for a presumed minor win in terms of bounds checking overhead 🙂

Reply:
Or for that matter, reading split_first's implmentation with slice patterns - but I think that's a much more recent feature `if let [first, tail @ ..] = self { Some((first, tail)) } else { None }`  but it's good to read and use elsewhere if not here.